core/src/main/java/com/linecorp/armeria/internal/common/stream/EmptyFixedStreamMessage.java

```diff
@@ -51,4 +83,19 @@ protected final List<T> drainAll(boolean withPooledObjects) {
 
     @Override
     public void request(long n) {}
+
+    @Override
+    public void cancel() {
+        whenComplete().complete(null);
```

Comment:
Question) Is there a reason `cancel()`, `abort()` are completed (instead of `completeExceptionally`?)

Reply:
Actually, since there aren't any objects I guess we can consider the message as already complete. Ignore this comment 😅

Reply:
>  I guess we can consider the message as already complete. 

+1
Also, it's weird that `StreamMessage.subscribe(...)` isn't completed with any exception when `whenComplete()` is not. 😄